packages/prettier-plugin-java/src/printers/expressions.ts

```diff
@@ -324,10 +321,50 @@ export class ExpressionsPrettierVisitor extends BaseCstPrettierPrinter {
     params?: { addParenthesisToWrapStatement?: boolean }
   ) {
     const countMethodInvocation = isUniqueMethodInvocation(ctx.primarySuffix);
+    const newExpression =
+      ctx.primaryPrefix[0].children.newExpression?.[0].children;
+    const arrayCreationExpression =
+      newExpression?.arrayCreationExpression?.[0].children;
+    const classInstanceCreationExpression =
+      newExpression?.unqualifiedClassInstanceCreationExpression?.[0].children;
+    const isBreakableNewExpression =
+      countMethodInvocation <= 1 &&
+      [
+        arrayCreationExpression?.classOrInterfaceType?.[0].children.classType[0]
+          .children.typeArguments,
+        arrayCreationExpression?.arrayCreationExplicitInitSuffix?.[0].children
+          .arrayInitializer[0].children.variableInitializerList,
+        classInstanceCreationExpression?.classOrInterfaceTypeToInstantiate[0]
+          .children.typeArgumentsOrDiamond?.[0].children.typeArguments,
+        classInstanceCreationExpression?.argumentList
+      ].some(breakablePart => breakablePart);
```

Comment:
We maybe could extract a isBreakableNewExpression function to lighten the cognitive load on this function ?